test/modules/TestResultCoordinator/Reports/ErrorReportMetadata.cs

```diff
@@ -2,13 +2,17 @@
 namespace TestResultCoordinator.Reports
 {
     using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
 
     class ErrorReportMetadata : ITestReportMetadata
     {
-        public ErrorReportMetadata()
+        public ErrorReportMetadata(string testDescription)
         {
+            this.TestDescription = Preconditions.CheckNonWhiteSpace(testDescription, nameof(testDescription));
```

Comment:
Since you are adding code commonly used for all reportMetadata, I think it is better to create an abstract class and implements constructor, TestDescription property and ToString() method.

Reply:
Good suggestion. Will fix.